docs/configuration.md

```diff
@@ -3134,3 +3134,119 @@ The stage level scheduling feature allows users to specify task and executor res
 This is only available for the RDD API in Scala, Java, and Python.  It is available on YARN and Kubernetes when dynamic allocation is enabled. See the [YARN](running-on-yarn.html#stage-level-scheduling-overview) page or [Kubernetes](running-on-kubernetes.html#stage-level-scheduling-overview) page for more implementation details.
 
 See the `RDD.withResources` and `ResourceProfileBuilder` API's for using this feature. The current implementation acquires new executors for each `ResourceProfile`  created and currently has to be an exact match. Spark does not try to fit tasks into an executor that require a different ResourceProfile than the executor was created with. Executors that are not in use will idle timeout with the dynamic allocation logic. The default configuration for this feature is to only allow one ResourceProfile per stage. If the user associates more then 1 ResourceProfile to an RDD, Spark will throw an exception by default. See config `spark.scheduler.resource.profileMergeConflicts` to control that behavior. The current merge strategy Spark implements when `spark.scheduler.resource.profileMergeConflicts` is enabled is a simple max of each resource within the conflicting ResourceProfiles. Spark will create a new ResourceProfile with the max of each of the resources.
+
+# Push-based shuffle overview
+
+Push-based shuffle is an improved shuffle architecture that optimizes the reliability and performance of the shuffle step in Spark. Complementing the existing shuffle mechanism, push-based shuffle takes a best-effort approach to push the shuffle blocks generated by the map tasks to remote shuffle services to be merged per shuffle partition. When the reduce tasks start running, they fetch a combination of the merged shuffle partitions and some of the original shuffle blocks to get their input data. As a result, push-based shuffle converts shuffle services’ small random disk reads into large sequential reads. The reduce tasks are also scheduled with locality preferences of the locations of their corresponding merged shuffle partitions, which helps to significantly improve shuffle fetch data locality.
+
+<p> <b> Currently push-based shuffle is only supported for Spark on YARN. </b></p>
```

Comment:
`.... supported for Spark on YARN with external shuffle service.`?

Reply:
Shall we also recommend the appropriate scenario here? For example, the workload with the high load of disk I/O. AFAIK, the push-based shuffle doesn't perform well on small/fast queries yet.

Reply:
Ideally once we have https://issues.apache.org/jira/browse/SPARK-33701 (Adaptive shuffle merge finalization) fixed, small/fast queries should perform roughly the same with the existing shuffle mechanism. Do you think we should document it for now and once SPARK-33701 we can remove/reword it accordingly. Thoughts?

Reply:
We are not targeting SPARK-33701 for 3.2 right ? Given that, let us document scenarios for 3.2 - for 3.3, SPARK-33701 would be an enhancement at which point 3.3 documentation would be enhanced to deal with the changes.

Reply:
In that case, I think we should mention this improvement as a work in progress in the documentation to not misguide users.

Reply:
Makes sense. Added this info in the Overview section.